File: frontend/lib/data/models/submission.dart
import 'package:hive/hive.dart';

part 'submission.g.dart';

@HiveType(typeId: 2)
class Submission extends HiveObject {
  @HiveField(0)
  final String taskId;

  @HiveField(1)
  final String userId;

  @HiveField(2)
  final List<String> beforeFiles;

  @HiveField(3)
  final List<String> afterFiles;

  @HiveField(4)
  final String? description;

  @HiveField(5)
  final DateTime createdAt;

  @HiveField(6)
  String status; // pending, approved, rejected

  @HiveField(7)
  final List<String> beforeFileNames;

  @HiveField(8)
  final List<String> afterFileNames;

  Submission({
    required this.taskId,
    required this.userId,
    required this.beforeFiles,
    required this.afterFiles,
    this.description,
    required this.status,
    required this.createdAt,
    this.beforeFileNames = const [],
    this.afterFileNames = const [],
  });
}

File: backend/models/Submission.js
const mongoose = require('mongoose');

const submissionSchema = new mongoose.Schema({
  task: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Task',
    required: true
  },
  employee: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  organizationId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Organization',
    required: true,
    index: true,
  },
  taskTitle: {
    type: String,
    required: true
  },
  employeeName: {
    type: String,
    required: true
  },
  beforeImage: {
    type: String, // Legacy S3 URL
    required: false
  },
  afterImage: {
    type: String, // Legacy S3 URL
    required: false
  },
  beforeFiles: [{
    type: String
  }],
  beforeFileNames: [{
    type: String
  }],
  afterFiles: [{
    type: String
  }],
  afterFileNames: [{
    type: String
  }],
  description: {
    type: String,
    trim: true
  },
  status: {
    type: String,
    enum: ['pending', 'approved', 'rejected'],
    default: 'pending'
  },
  adminFeedback: {
    type: String,
    trim: true
  },
  comments: [{
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
    text: { type: String, required: true },
    createdAt: { type: Date, default: Date.now }
  }]
}, { timestamps: true });

module.exports = mongoose.model('Submission', submissionSchema);
